python/dgl/distributed/nn/pytorch/sparse_emb.py

```diff
@@ -0,0 +1,194 @@
+        self._trace = []
+        self._name = name
+        self._num_embeddings = num_embeddings
+        self._embedding_dim = embedding_dim
+
+        # Check whether it is multi-gpu/distributed training or not
+        if th.distributed.is_initialized():
+            self._rank = th.distributed.get_rank()
+            self._world_size = th.distributed.get_world_size()
+        else:
+            assert 'th.distributed shoud be initialized'
+        self._optm_state = None # track optimizer state
+        self._part_policy = part_policy
+
+    def __call__(self, idx, device=th.device('cpu')):
+        """
+        node_ids : th.tensor
+            Index of the embeddings to collect.
+        device : th.device
+            Target device to put the collected embeddings.
+
+        Returns
+        -------
+        Tensor
+            The requested node embeddings
+        """
+        idx = utils.toindex(idx).tousertensor()
+        emb = self._tensor[idx].to(device, non_blocking=True)
+        if F.is_recording():
+            emb = F.attach_grad(emb)
+            self._trace.append((idx.to(device, non_blocking=True), emb))
+        return emb
+
+    def reset_trace(self):
+        '''Reset the traced data.
+        '''
+        self._trace = []
+
+    @property
+    def part_policy(self):
+        """Return the partition policy
+
+        Returns
+        -------
+        PartitionPolicy
+            partition policy
+        """
+        return self._part_policy
+
+    @property
+    def name(self):
+        """Return the name of the embeddings
+
+        Returns
+        -------
+        str
+            The name of the embeddings
+        """
+        return self._tensor.tensor_name
+
+    @property
+    def kvstore(self):
+        """Return the kvstore client
+
+        Returns
+        -------
+        KVClient
+            The kvstore client
+        """
+        return self._tensor.kvstore
+
+    @property
+    def num_embeddings(self):
+        """Return the number of embeddings
+
+        Returns
+        -------
+        int
+            The number of embeddings
+        """
+        return self._num_embeddings
+
+    @property
+    def embedding_dim(self):
+        """Return the dimension of embeddings
+
+        Returns
+        -------
+        int
+            The dimension of embeddings
+        """
+        return self._embedding_dim
+
+    def set_optm_state(self, state):
+        """Store the optimizer related state tensor.
+
+        Parameters
+        ----------
+        state : tuple of torch.Tensor
+            Optimizer related state.
+        """
+        self._optm_state = state
+
+    @property
+    def optm_state(self):
+        """Return the optimizer related state tensor.
+
+        Returns
+        -------
+        tuple of torch.Tensor
+            The optimizer related state.
+        """
+        return self._optm_state
+
+    @property
+    def weight(self):
```

Comment:
i think the non-distributed version of NodeEmbedding has a field called `emb_tensor`. do you want to be consistent?

Reply:
non-distributed version now have emb_tensor and weight and we will deprecate emb_tensor later.